### PR TITLE
RavenDB-7889 fixes after PR

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 1]
```diff
@@ -318,6 +331,37 @@ private static IEnumerable<OngoingTask> CollectEtlTasks(DatabaseRecord databaseR
...
+        {
+            string error = null;
+            db = null;
+            try
+            {
+                if (store.DatabasesLandlord.DatabasesCache.TryGetValue(name, out var task))
+                {
+                    if (task == null)
+                    {
+                        throw new DatabaseLoadFailureException("Database task is 'null'.");
```
Comment (ayende) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:344]: Which database task?

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 2]
```diff
@@ -318,6 +331,37 @@ private static IEnumerable<OngoingTask> CollectEtlTasks(DatabaseRecord databaseR
...
+                    }
+                    if (task.IsCanceled)
+                    {
+                        throw new TaskCanceledException("Database task was canceled.");
```
Comment (ayende) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:348]: Which database task?

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 3]
```diff
@@ -318,6 +331,37 @@ private static IEnumerable<OngoingTask> CollectEtlTasks(DatabaseRecord databaseR
...
+                    }
+                    if (task.IsFaulted)
+                    {
+                        throw new DatabaseLoadFailureException("Database task is faulted.", task.Exception);
```
Comment (ayende) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:352]: Which ?

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 4]
```diff
@@ -318,6 +331,37 @@ private static IEnumerable<OngoingTask> CollectEtlTasks(DatabaseRecord databaseR
...
+                    }
+                    db = task.Result;
```
Comment (ayende) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:354]: And if this is loaded? You'll hang until it is done, can be a LONG time

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Web/System/OngoingTasksHandler.cs:344]: those errors will be caught by the 'catch' and add the database name to the error

**Pull request merged**